docs/7683-discussion.md

```diff
@@ -0,0 +1,60 @@
+# 7683 (Open Intents Framework) - Goals & Issues
+
+## Context
+There’s general consensus around pushing 7683 as an generic intents standard. In this document, we’ll attempt to clarify some of the goals of the effort and questions around the interface. Feel free to add your team’s POV to the doc!
+
+## Goals
+Are we all in agreement on the high level goals of the open intents framework (OIF)?
+
+* Should the generic solver implementation in the OIF work out of the box for any 7683 intents protocol (w/ no additional changes required)?
+* Should the Base7683 implementation work out of the box for any intents protocol to make them 7683 compliant?
+* Is the end goal to have a single 7683 contract impl deployed to each chain that protocols can plug into or is it to have each intent protocol inherit from the Base7683 contract and create/deploy their own implementations?
+Is it worth aiming to arrive at a version that could eventually be adopted as a precompile to standardize interactions? 
+
+## Issues
+
+* Deposits.
+  * Across’s implementation:
+    * ERC7683OrderDepositor exposes an internal `_callDeposit` method that lets the underlying protocol define the deposit mechanics.
+	* ERC7683OrderDepositorExternal translates the `_callDeposit()` call into an Across’s own deposit impl in their “spoke pool”.
+	* `_callDeposit` API exposes Across specific params such as exclusivityPeriod that are not defined in the 7683 spec. 
+  * OIF implementation:
+    * Open methods implement the deposit logic. This means that protocol specific deposit logic is currently not supported.
+  * Resource lock impls generally require deposits to be made within the resource lock itself. This makes them incompatible with OIF’s current impl. 
+* Order data encoding.
+  * Across’s impl has many Across specific fields in their order data. The OIF impl does leave _fillOrder up to the inheriting contract to impl. 
+  * This means that we’ll need a separate solver impl per intents protocol. Is this okay? Do we want to standardize some of the order data fields?
```

Comment:
I think it is easier to have reference solver modules that abstract some of these differences than to try to standardize everything up front. The nice thing about 7683 contracts is that they can be stateless, so the interface can be additive over time